src/marshmallow/fields.py

```diff
@@ -1295,19 +1295,36 @@ def _deserialize(self, value, attr, data, **kwargs):
 class Time(Field):
     """ISO8601-formatted time string.
 
+    :param format: ISO8601 strftime format of the value.
+        If not specified, defaults to built-in ``isoformat`` de/serialization.
     :param kwargs: The same keyword arguments that :class:`Field` receives.
     """
 
     #: Default error messages.
     default_error_messages = {
         "invalid": "Not a valid time.",
+        "iso_format": "Not a valid ISO8601 format",
         "format": '"{input}" cannot be formatted as a time.',
     }
 
+    VALID_ISO_STRFTIME_FORMATS: typing.List[str] = [
+        "%H",
+        "%H:%M",
+        "%H:%M:%S",
+        "%H:%M:%S.%f",
+    ]
+
+    def __init__(self, format: typing.Optional[str] = None, **kwargs):
+        super().__init__(**kwargs)
+        self.format = format
+
+        if self.format and self.format not in Time.VALID_ISO_STRFTIME_FORMATS:
```

Comment:
thought: I'm not sure we should constrain the set of strftime formats; `fields.DateTime` does not do this, and we should keep this consistent.